contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+        public
+        view
+        returns (uint256 validationCode)
+    {
+        if (makerOrder.signer == address(0)) return MAKER_SIGNER_IS_NULL_SIGNER;
+
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", _DOMAIN_SEPARATOR, makerOrder.hash()));
+
+        if (!Address.isContract(makerOrder.signer)) {
+            uint256 response = _validateEOA(digest, makerOrder.signer, makerOrder.v, makerOrder.r, makerOrder.s);
+            if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        } else {
+            uint256 response = _validateERC1271(digest, makerOrder.signer, makerOrder.v, makerOrder.r, makerOrder.s);
+            if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        }
+    }
+
+    /**
+     * @notice Check validity of whitelists
+     * @param makerOrder maker order struct
+     * @return validationCode validation code
+     */
+    function checkValidityWhitelists(OrderTypes.MakerOrder calldata makerOrder)
+        public
+        view
+        returns (uint256 validationCode)
+    {
+        // Verify whether the currency is whitelisted
+        if (!currencyManager.isCurrencyWhitelisted(makerOrder.currency)) return CURRENCY_NOT_WHITELISTED;
+
+        // Verify whether strategy can be executed
+        if (!executionManager.isStrategyWhitelisted(makerOrder.strategy)) return STRATEGY_NOT_WHITELISTED;
+    }
+
+    /**
+     * @notice Check validity of min percentage to ask
+     * @param makerOrder maker order struct
+     * @return validationCode validation code
+     */
+    function checkValidityMinPercentageToAsk(OrderTypes.MakerOrder calldata makerOrder)
+        public
+        view
+        returns (uint256 validationCode)
+    {
+        // Return if order is bid since there is no protection for minPercentageToAsk
+        if (!makerOrder.isOrderAsk) return ORDER_EXPECTED_TO_BE_VALID;
+
+        uint256 finalSellerAmount = makerOrder.price;
+        uint256 protocolFee = (makerOrder.price * IExecutionStrategy(makerOrder.strategy).viewProtocolFee()) / 10000;
+        finalSellerAmount -= protocolFee;
+
+        if ((finalSellerAmount * 10000) < (makerOrder.minPercentageToAsk * makerOrder.price))
+            return MIN_NET_RATIO_ABOVE_PROTOCOL_FEE;
+
+        (address receiver, uint256 royaltyAmount) = royaltyFeeRegistry.royaltyInfo(
+            makerOrder.collection,
+            makerOrder.price
+        );
+
+        if ((receiver != address(0)) && (royaltyAmount != 0)) {
+            // Royalty registry logic
+            finalSellerAmount -= royaltyAmount;
+            if ((finalSellerAmount * 10000) < (makerOrder.minPercentageToAsk * makerOrder.price))
+                return MIN_NET_RATIO_ABOVE_ROYALTY_FEE_REGISTRY_AND_PROTOCOL_FEE;
+        } else {
+            // ERC2981 logic
+            if (IERC165(makerOrder.collection).supportsInterface(0x2a55205a)) {
+                (bool answer, bytes memory data) = makerOrder.collection.staticcall(
```

Comment:
Nothing wrong at all, but why the variable name `answer` instead of what is usually named `success`?

Reply:
Adjusted